### PR TITLE
Integrate tracer with task manager

### PR DESCRIPTION
**Pull request by DaveCTurner**

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/tasks/TaskTracer.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    public void onTaskUnregistered(Task task) {
+        for (TracingPlugin.Tracer tracer : tracers) {
+            try {
+                tracer.onTaskUnregistered(task);
+            } catch (Exception e) {
+                assert false : e;
+                logger.warn(
+                    new ParameterizedMessage(
+                        "task tracing listener [{}] failed on registration of task [{}][{}]",
```
Comment (tlrx) [inline server/src/main/java/org/elasticsearch/tasks/TaskTracer.java:58]:
```suggestion
                        "task tracing listener [{}] failed on unregistration of task [{}][{}]",
```

**Review by tlrx — APPROVED** [review]
LGTM

**Comment by DaveCTurner** [comment]
@elasticmachine please run elasticsearch-ci/part-1

**Comment by DaveCTurner** [comment]
@elasticmachine please run elasticsearch-ci/part-2

**Comment by DaveCTurner** [comment]
@elasticmachine please run elasticsearch-ci/part-2

**Pull request merged**